app/plugins/Wallet/js/lifecycle.js

```diff
@@ -91,14 +91,17 @@ function start() {
 	// DEVTOOL: uncomment to bring up devtools on plugin view
 	// IPC.sendToHost('devtools');
 	
-	// Need to check if wallet's unencrypted
-	IPC.sendToHost('api-call', '/wallet', 'on-opened');
+	// Need to check if wallet's unencrypted, keep checking until we get a response
+	updating = setInterval(IPC.sendToHost, 500, 'api-call', '/wallet', 'on-opened');
```

Comment:
I would do this in a slightly different way. You're using the same variable, updating, that the update function uses and that's confusing its purpose. Make a new variable called `starting`?